lib_ecstasy/src/main/x/ecstasy/text/Char.x

```diff
@@ -182,7 +186,11 @@ const Char(UInt32 codepoint)
     String add(String s) = new StringBuffer(1 + s.size).add(this).addAll(s).toString();
 
     @Op("-")
-    Char sub(UInt32 n) = new Char(codepoint - n);
+    Char sub(Int n) {
+        assert:bounds -(0x10FFFF-codepoint) <= n <= codepoint as
+                $"Subtracting {n} from character code-point {codepoint} would exceed the Unicode range";
```

Comment:
The XTC convention for the rest of the tree seems to prescribe space around binary operators.